examples/shors_code.py

```diff
@@ -0,0 +1,118 @@
+
+    def encode(self):
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[3])])
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[6])])
+        yield cirq.ops.Moment([
+            cirq.H(self.physical_qubits[0]),
+            cirq.H(self.physical_qubits[3]),
+            cirq.H(self.physical_qubits[6])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[1]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[4]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[7])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[2]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[5]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[8])
+        ])
+
+    def apply_gate(self, gate: cirq.Gate, pos: int):
+        if pos > self.num_physical_qubits:
+            raise IndexError
+        else:
+            return gate(self.physical_qubits[pos])
+
+    def correct(self):
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[1]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[4]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[7])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CNOT(self.physical_qubits[0], self.physical_qubits[2]),
+            cirq.CNOT(self.physical_qubits[3], self.physical_qubits[5]),
+            cirq.CNOT(self.physical_qubits[6], self.physical_qubits[8])
+        ])
+        yield cirq.ops.Moment([
+            cirq.CCNOT(self.physical_qubits[1], self.physical_qubits[2],
+                       self.physical_qubits[0]),
+            cirq.CCNOT(self.physical_qubits[4], self.physical_qubits[5],
+                       self.physical_qubits[3]),
+            cirq.CCNOT(self.physical_qubits[7], self.physical_qubits[8],
+                       self.physical_qubits[6])
+        ])
+        yield cirq.ops.Moment([
+            cirq.H(self.physical_qubits[0]),
+            cirq.H(self.physical_qubits[3]),
+            cirq.H(self.physical_qubits[6])
+        ])
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[3])])
+        yield cirq.ops.Moment(
+            [cirq.CNOT(self.physical_qubits[0], self.physical_qubits[6])])
+        yield cirq.ops.Moment([
+            cirq.CCNOT(self.physical_qubits[3], self.physical_qubits[6],
+                       self.physical_qubits[0])
+        ])
+
+    def measure(self):
+        for i in range(self.num_physical_qubits):
+            yield cirq.measure(self.physical_qubits[i])
+
+
+if __name__ == '__main__':
+    # coverage: ignore
+    #create circuit
+    mycode1 = OneQubitShorsCode()
+    my_circuit1 = cirq.Circuit(mycode1.encode())
```

Comment:
What do you think about this: 

```
    # coverage: ignore

    #create circuit with 9 physical qubits
    code = OneQubitShorsCode()

    circuit = cirq.Circuit(code.apply_gate(cirq.X ** (1 / 4), 0))
    print(cirq.dirac_notation(circuit.final_state_vector(initial_state=0)))

    circuit += cirq.Circuit(code.encode())
    print(cirq.dirac_notation(circuit.final_state_vector(initial_state=0)))

    #create error
    circuit += cirq.Circuit(
        code.apply_gate(cirq.X,
                           random.randint(0, code.num_physical_qubits - 1)))
    print(cirq.dirac_notation(circuit.final_state_vector(initial_state=0)))

    #correct error and decode
    circuit += cirq.Circuit(code.correct())
    print(cirq.dirac_notation(circuit.final_state_vector(initial_state=0)))

```

Reply:
1 Yeah it makes sense to delete measure method.
2 I put down Shor95 as reference.
3 Looks good! Merging the newly added final_state_vector is blocked right now. It seems I need to address the requested change first.

Reply:
What is the purpose of cirq.X**(1/4) before encode?

Reply:
It is to create an interesting superposition - in the end any <img src="https://render.githubusercontent.com/render/math?math=%5Clvert%20%5Cpsi%20%5Crangle%20%3D%20%5Calpha%20%5Clvert%200%20%5Crangle%20%2B%20%5Cbeta%20%5Clvert%201%20%5Crangle%20"> should be preserved over an error channel of up to one phase and bit-flip errors or the combination of them. We can show at the end that after decoding the first qubit is still in the same state. 

While you were showing this with |0> I think a more interesting superstition can drive the message home even better.

Reply:
Great idea! 

Using dirac notation allows us to show superposition well.

Reply:
Just pushed the changes. Also changed the test file.